{"cells":[{"cell_type":"markdown","metadata":{},"source":["#### imports"]},{"cell_type":"code","execution_count":7,"metadata":{},"outputs":[],"source":["import itertools\n","import numpy as np\n","import matplotlib.pyplot as plt\n","from matplotlib.ticker import NullFormatter\n","import pandas as pd\n","import numpy as np\n","import matplotlib.ticker as ticker\n","from sklearn import preprocessing\n","%matplotlib inline"]},{"cell_type":"markdown","metadata":{},"source":["# load the data"]},{"cell_type":"code","execution_count":8,"metadata":{},"outputs":[],"source":["#!wget -O loan_train.csv https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/ML0101ENv3/labs/loan_train.csv"]},{"cell_type":"markdown","metadata":{},"source":["#### load csv into dataframe"]},{"cell_type":"code","execution_count":9,"metadata":{},"outputs":[{"data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>Unnamed: 0</th>\n","      <th>Unnamed: 0.1</th>\n","      <th>loan_status</th>\n","      <th>Principal</th>\n","      <th>terms</th>\n","      <th>effective_date</th>\n","      <th>due_date</th>\n","      <th>age</th>\n","      <th>education</th>\n","      <th>Gender</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>9/8/2016</td>\n","      <td>10/7/2016</td>\n","      <td>45</td>\n","      <td>High School or Below</td>\n","      <td>male</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>2</td>\n","      <td>2</td>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>9/8/2016</td>\n","      <td>10/7/2016</td>\n","      <td>33</td>\n","      <td>Bechalor</td>\n","      <td>female</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>3</td>\n","      <td>3</td>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>15</td>\n","      <td>9/8/2016</td>\n","      <td>9/22/2016</td>\n","      <td>27</td>\n","      <td>college</td>\n","      <td>male</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>4</td>\n","      <td>4</td>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>9/9/2016</td>\n","      <td>10/8/2016</td>\n","      <td>28</td>\n","      <td>college</td>\n","      <td>female</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>6</td>\n","      <td>6</td>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>9/9/2016</td>\n","      <td>10/8/2016</td>\n","      <td>29</td>\n","      <td>college</td>\n","      <td>male</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["   Unnamed: 0  Unnamed: 0.1 loan_status  Principal  terms effective_date  \\\n","0           0             0     PAIDOFF       1000     30       9/8/2016   \n","1           2             2     PAIDOFF       1000     30       9/8/2016   \n","2           3             3     PAIDOFF       1000     15       9/8/2016   \n","3           4             4     PAIDOFF       1000     30       9/9/2016   \n","4           6             6     PAIDOFF       1000     30       9/9/2016   \n","\n","    due_date  age             education  Gender  \n","0  10/7/2016   45  High School or Below    male  \n","1  10/7/2016   33              Bechalor  female  \n","2  9/22/2016   27               college    male  \n","3  10/8/2016   28               college  female  \n","4  10/8/2016   29               college    male  "]},"execution_count":9,"metadata":{},"output_type":"execute_result"}],"source":["df = pd.read_csv('loan_train.csv')\n","df.head()"]},{"cell_type":"code","execution_count":10,"metadata":{},"outputs":[{"data":{"text/plain":["(346, 10)"]},"execution_count":10,"metadata":{},"output_type":"execute_result"}],"source":["df.shape"]},{"cell_type":"markdown","metadata":{},"source":["# clean the data"]},{"cell_type":"markdown","metadata":{},"source":["#### remove unamed"]},{"cell_type":"code","execution_count":11,"metadata":{},"outputs":[{"data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>loan_status</th>\n","      <th>Principal</th>\n","      <th>terms</th>\n","      <th>effective_date</th>\n","      <th>due_date</th>\n","      <th>age</th>\n","      <th>education</th>\n","      <th>Gender</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>9/8/2016</td>\n","      <td>10/7/2016</td>\n","      <td>45</td>\n","      <td>High School or Below</td>\n","      <td>male</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>9/8/2016</td>\n","      <td>10/7/2016</td>\n","      <td>33</td>\n","      <td>Bechalor</td>\n","      <td>female</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>15</td>\n","      <td>9/8/2016</td>\n","      <td>9/22/2016</td>\n","      <td>27</td>\n","      <td>college</td>\n","      <td>male</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>9/9/2016</td>\n","      <td>10/8/2016</td>\n","      <td>28</td>\n","      <td>college</td>\n","      <td>female</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>9/9/2016</td>\n","      <td>10/8/2016</td>\n","      <td>29</td>\n","      <td>college</td>\n","      <td>male</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["  loan_status  Principal  terms effective_date   due_date  age  \\\n","0     PAIDOFF       1000     30       9/8/2016  10/7/2016   45   \n","1     PAIDOFF       1000     30       9/8/2016  10/7/2016   33   \n","2     PAIDOFF       1000     15       9/8/2016  9/22/2016   27   \n","3     PAIDOFF       1000     30       9/9/2016  10/8/2016   28   \n","4     PAIDOFF       1000     30       9/9/2016  10/8/2016   29   \n","\n","              education  Gender  \n","0  High School or Below    male  \n","1              Bechalor  female  \n","2               college    male  \n","3               college  female  \n","4               college    male  "]},"execution_count":11,"metadata":{},"output_type":"execute_result"}],"source":["df = df.drop([\"Unnamed: 0\", \"Unnamed: 0.1\"], axis=1)\n","df.head()"]},{"cell_type":"markdown","metadata":{},"source":["#### convert date"]},{"cell_type":"code","execution_count":12,"metadata":{},"outputs":[{"data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>loan_status</th>\n","      <th>Principal</th>\n","      <th>terms</th>\n","      <th>effective_date</th>\n","      <th>due_date</th>\n","      <th>age</th>\n","      <th>education</th>\n","      <th>Gender</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>2016-09-08</td>\n","      <td>2016-10-07</td>\n","      <td>45</td>\n","      <td>High School or Below</td>\n","      <td>male</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>2016-09-08</td>\n","      <td>2016-10-07</td>\n","      <td>33</td>\n","      <td>Bechalor</td>\n","      <td>female</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>15</td>\n","      <td>2016-09-08</td>\n","      <td>2016-09-22</td>\n","      <td>27</td>\n","      <td>college</td>\n","      <td>male</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>2016-09-09</td>\n","      <td>2016-10-08</td>\n","      <td>28</td>\n","      <td>college</td>\n","      <td>female</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>2016-09-09</td>\n","      <td>2016-10-08</td>\n","      <td>29</td>\n","      <td>college</td>\n","      <td>male</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["  loan_status  Principal  terms effective_date   due_date  age  \\\n","0     PAIDOFF       1000     30     2016-09-08 2016-10-07   45   \n","1     PAIDOFF       1000     30     2016-09-08 2016-10-07   33   \n","2     PAIDOFF       1000     15     2016-09-08 2016-09-22   27   \n","3     PAIDOFF       1000     30     2016-09-09 2016-10-08   28   \n","4     PAIDOFF       1000     30     2016-09-09 2016-10-08   29   \n","\n","              education  Gender  \n","0  High School or Below    male  \n","1              Bechalor  female  \n","2               college    male  \n","3               college  female  \n","4               college    male  "]},"execution_count":12,"metadata":{},"output_type":"execute_result"}],"source":["df['due_date'] = pd.to_datetime(df['due_date'])\n","df['effective_date'] = pd.to_datetime(df['effective_date'])\n","df.head()"]},{"cell_type":"markdown","metadata":{},"source":["#### loan_status column is what we want to predict (target)"]},{"cell_type":"code","execution_count":13,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["['PAIDOFF' 'COLLECTION']\n","PAIDOFF       260\n","COLLECTION     86\n","Name: loan_status, dtype: int64\n"]}],"source":["print(df['loan_status'].unique())\n","print(df['loan_status'].value_counts())"]},{"cell_type":"markdown","metadata":{},"source":["#### convert male to 0 and female to 1"]},{"cell_type":"code","execution_count":14,"metadata":{},"outputs":[{"data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>loan_status</th>\n","      <th>Principal</th>\n","      <th>terms</th>\n","      <th>effective_date</th>\n","      <th>due_date</th>\n","      <th>age</th>\n","      <th>education</th>\n","      <th>Gender</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>2016-09-08</td>\n","      <td>2016-10-07</td>\n","      <td>45</td>\n","      <td>High School or Below</td>\n","      <td>0</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>2016-09-08</td>\n","      <td>2016-10-07</td>\n","      <td>33</td>\n","      <td>Bechalor</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>15</td>\n","      <td>2016-09-08</td>\n","      <td>2016-09-22</td>\n","      <td>27</td>\n","      <td>college</td>\n","      <td>0</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>2016-09-09</td>\n","      <td>2016-10-08</td>\n","      <td>28</td>\n","      <td>college</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>PAIDOFF</td>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>2016-09-09</td>\n","      <td>2016-10-08</td>\n","      <td>29</td>\n","      <td>college</td>\n","      <td>0</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["  loan_status  Principal  terms effective_date   due_date  age  \\\n","0     PAIDOFF       1000     30     2016-09-08 2016-10-07   45   \n","1     PAIDOFF       1000     30     2016-09-08 2016-10-07   33   \n","2     PAIDOFF       1000     15     2016-09-08 2016-09-22   27   \n","3     PAIDOFF       1000     30     2016-09-09 2016-10-08   28   \n","4     PAIDOFF       1000     30     2016-09-09 2016-10-08   29   \n","\n","              education  Gender  \n","0  High School or Below       0  \n","1              Bechalor       1  \n","2               college       0  \n","3               college       1  \n","4               college       0  "]},"execution_count":14,"metadata":{},"output_type":"execute_result"}],"source":["df['Gender'].replace(to_replace=['male','female'], value=[0,1],inplace=True)\n","df.head()"]},{"cell_type":"markdown","metadata":{},"source":["#### set a column for categorical variable with 0 or 1"]},{"cell_type":"code","execution_count":15,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["['High School or Below' 'Bechalor' 'college' 'Master or Above']\n","High School or Below    151\n","college                 149\n","Bechalor                 44\n","Master or Above           2\n","Name: education, dtype: int64\n"]}],"source":["print(df['education'].unique())\n","print(df['education'].value_counts())"]},{"cell_type":"code","execution_count":16,"metadata":{},"outputs":[{"data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>Principal</th>\n","      <th>terms</th>\n","      <th>age</th>\n","      <th>Gender</th>\n","      <th>Bechalor</th>\n","      <th>High School or Below</th>\n","      <th>college</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>45</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>1</td>\n","      <td>0</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>33</td>\n","      <td>1</td>\n","      <td>1</td>\n","      <td>0</td>\n","      <td>0</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>1000</td>\n","      <td>15</td>\n","      <td>27</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>28</td>\n","      <td>1</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>1000</td>\n","      <td>30</td>\n","      <td>29</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>1</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["   Principal  terms  age  Gender  Bechalor  High School or Below  college\n","0       1000     30   45       0         0                     1        0\n","1       1000     30   33       1         1                     0        0\n","2       1000     15   27       0         0                     0        1\n","3       1000     30   28       1         0                     0        1\n","4       1000     30   29       0         0                     0        1"]},"execution_count":16,"metadata":{},"output_type":"execute_result"}],"source":["# keep non categorical columns\n","Feature = df[['Principal','terms','age','Gender']]\n","# set a column for each categorical value\n","Feature = pd.concat([Feature,pd.get_dummies(df['education'])], axis=1)\n","# drop \"Master or above\" column, too few values\n","Feature.drop(['Master or Above'], axis = 1,inplace=True)\n","Feature.head()"]},{"cell_type":"markdown","metadata":{},"source":["#### set X and Y for classification"]},{"cell_type":"code","execution_count":17,"metadata":{},"outputs":[],"source":["X = Feature\n","y = df['loan_status'].values"]},{"cell_type":"markdown","metadata":{},"source":["#### Normalize Data"]},{"cell_type":"code","execution_count":18,"metadata":{},"outputs":[],"source":["X = preprocessing.StandardScaler().fit(X).transform(X)"]},{"cell_type":"markdown","metadata":{},"source":["# Classification"]},{"cell_type":"markdown","metadata":{},"source":["#### split the dataset"]},{"cell_type":"code","execution_count":19,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["Train set: (276, 7) (276,)\n","Test set: (70, 7) (70,)\n"]}],"source":["from sklearn.model_selection import train_test_split\n","\n","X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)\n","print ('Train set:', X_train.shape,  y_train.shape)\n","print ('Test set:', X_test.shape,  y_test.shape)"]},{"cell_type":"markdown","metadata":{},"source":["## KNN"]},{"cell_type":"code","execution_count":20,"metadata":{},"outputs":[{"data":{"text/plain":["0.5571428571428572"]},"execution_count":20,"metadata":{},"output_type":"execute_result"}],"source":["from sklearn.neighbors import KNeighborsClassifier\n","\n","# train Model and Predict with k = 3\n","kNN_model = KNeighborsClassifier(n_neighbors=3)\n","kNN_model.fit(X_train, y_train)\n","kNN_model.score(X_test, y_test)"]},{"cell_type":"code","execution_count":21,"metadata":{},"outputs":[{"name":"stderr","output_type":"stream","text":["c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 1157, in fit\n","    return self._fit(X)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 350, in _fit\n","    self._check_algorithm_metric()\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 332, in _check_algorithm_metric\n","    raise ValueError(\"Metric '%s' not valid. Use \"\n","ValueError: Metric 'euclidiean' not valid. Use sorted(sklearn.neighbors.VALID_METRICS['brute']) to get valid options. Metric can also be a callable function.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"]},{"data":{"text/plain":["GridSearchCV(cv=5, estimator=KNeighborsClassifier(),\n","             param_grid={'metric': ['euclidiean', 'manhattan'],\n","                         'n_neighbors': array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,\n","       18, 19])})"]},"execution_count":21,"metadata":{},"output_type":"execute_result"}],"source":["# use GridSearchCV to get best hyperparameters for KNN\n","from sklearn.model_selection import GridSearchCV\n","\n","param_grid = {'n_neighbors' : np.arange(1, 20), 'metric' : ['euclidiean', 'manhattan']}\n","\n","grid = GridSearchCV(KNeighborsClassifier(), param_grid, cv=5)\n","grid.fit(X_train, y_train)"]},{"cell_type":"code","execution_count":22,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["best score for KNN 0.7644805194805195\n","best hypermarameters for KNN {'metric': 'manhattan', 'n_neighbors': 17}\n"]}],"source":["print(\"best score for KNN\", grid.best_score_)\n","print(\"best hypermarameters for KNN\", grid.best_params_)"]},{"cell_type":"code","execution_count":23,"metadata":{},"outputs":[{"data":{"text/plain":["0.7"]},"execution_count":23,"metadata":{},"output_type":"execute_result"}],"source":["kNN_model = grid.best_estimator_\n","kNN_model.score(X_test, y_test)"]},{"cell_type":"markdown","metadata":{},"source":["## Decision tree"]},{"cell_type":"code","execution_count":24,"metadata":{},"outputs":[{"data":{"text/plain":["0.5285714285714286"]},"execution_count":24,"metadata":{},"output_type":"execute_result"}],"source":["from sklearn.tree import DecisionTreeClassifier\n","\n","DT_model = DecisionTreeClassifier()\n","DT_model.fit(X_train,y_train)\n","DT_model.score(X_test, y_test)"]},{"cell_type":"code","execution_count":25,"metadata":{},"outputs":[{"data":{"text/plain":["GridSearchCV(cv=5, estimator=DecisionTreeClassifier(),\n","             param_grid={'criterion': ['gini', 'entropy'],\n","                         'max_depth': array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,\n","       18, 19])})"]},"execution_count":25,"metadata":{},"output_type":"execute_result"}],"source":["param_grid = {'criterion' : ['gini', 'entropy'], 'max_depth' : np.arange(1, 20)}\n","\n","grid = GridSearchCV(DecisionTreeClassifier(), param_grid, cv=5)\n","grid.fit(X_train, y_train)"]},{"cell_type":"code","execution_count":26,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["best score for Decision Tree 0.7644805194805195\n","best hypermarameters for Decision Tree {'criterion': 'gini', 'max_depth': 1}\n"]}],"source":["print(\"best score for Decision Tree\", grid.best_score_)\n","print(\"best hypermarameters for Decision Tree\", grid.best_params_)"]},{"cell_type":"code","execution_count":27,"metadata":{},"outputs":[{"data":{"text/plain":["0.7"]},"execution_count":27,"metadata":{},"output_type":"execute_result"}],"source":["DT_model = grid.best_estimator_\n","DT_model.score(X_test, y_test)"]},{"cell_type":"markdown","metadata":{},"source":["## Support Vector Machine"]},{"cell_type":"code","execution_count":28,"metadata":{},"outputs":[{"data":{"text/plain":["0.7"]},"execution_count":28,"metadata":{},"output_type":"execute_result"}],"source":["from sklearn import svm\n","\n","SVM_model = svm.SVC()\n","SVM_model.fit(X_train, y_train)\n","SVM_model.score(X_test, y_test)"]},{"cell_type":"code","execution_count":29,"metadata":{},"outputs":[{"data":{"text/plain":["GridSearchCV(cv=5, estimator=SVC(),\n","             param_grid={'C': [0.1, 1, 10, 100, 1000],\n","                         'gamma': [1, 0.1, 0.01, 0.001, 0.0001],\n","                         'kernel': ['rbf']})"]},"execution_count":29,"metadata":{},"output_type":"execute_result"}],"source":["param_grid = {'C': [0.1, 1, 10, 100, 1000],\n","              'gamma': [1, 0.1, 0.01, 0.001, 0.0001],\n","              'kernel': ['rbf']}\n","\n","grid = GridSearchCV(svm.SVC(), param_grid, cv=5)\n","grid.fit(X_train, y_train)"]},{"cell_type":"code","execution_count":30,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["best score for SVM 0.7644805194805195\n","best hypermarameters for SVM {'C': 0.1, 'gamma': 1, 'kernel': 'rbf'}\n"]}],"source":["print(\"best score for SVM\", grid.best_score_)\n","print(\"best hypermarameters for SVM\", grid.best_params_)"]},{"cell_type":"code","execution_count":31,"metadata":{},"outputs":[{"data":{"text/plain":["0.7"]},"execution_count":31,"metadata":{},"output_type":"execute_result"}],"source":["SVM_model = grid.best_estimator_\n","SVM_model.score(X_test, y_test)"]},{"cell_type":"markdown","metadata":{},"source":["## Logistic Regression"]},{"cell_type":"code","execution_count":32,"metadata":{},"outputs":[{"data":{"text/plain":["0.7"]},"execution_count":32,"metadata":{},"output_type":"execute_result"}],"source":["from sklearn.linear_model import LogisticRegression\n","\n","LR_model = LogisticRegression()\n","LR_model.fit(X_train,y_train)\n","LR_model.score(X_test, y_test)"]},{"cell_type":"code","execution_count":33,"metadata":{},"outputs":[{"name":"stderr","output_type":"stream","text":["c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1304, in fit\n","    solver = _check_solver(self.solver, self.penalty, self.dual)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 442, in _check_solver\n","    raise ValueError(\"Solver %s supports only 'l2' or 'none' penalties, \"\n","ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1304, in fit\n","    solver = _check_solver(self.solver, self.penalty, self.dual)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 442, in _check_solver\n","    raise ValueError(\"Solver %s supports only 'l2' or 'none' penalties, \"\n","ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1304, in fit\n","    solver = _check_solver(self.solver, self.penalty, self.dual)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 442, in _check_solver\n","    raise ValueError(\"Solver %s supports only 'l2' or 'none' penalties, \"\n","ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1304, in fit\n","    solver = _check_solver(self.solver, self.penalty, self.dual)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 442, in _check_solver\n","    raise ValueError(\"Solver %s supports only 'l2' or 'none' penalties, \"\n","ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1304, in fit\n","    solver = _check_solver(self.solver, self.penalty, self.dual)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 442, in _check_solver\n","    raise ValueError(\"Solver %s supports only 'l2' or 'none' penalties, \"\n","ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1304, in fit\n","    solver = _check_solver(self.solver, self.penalty, self.dual)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 442, in _check_solver\n","    raise ValueError(\"Solver %s supports only 'l2' or 'none' penalties, \"\n","ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n","c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:548: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n","Traceback (most recent call last):\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 531, in _fit_and_score\n","    estimator.fit(X_train, y_train, **fit_params)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1304, in fit\n","    solver = _check_solver(self.solver, self.penalty, self.dual)\n","  File \"c:\\Users\\gerin\\anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 442, in _check_solver\n","    raise ValueError(\"Solver %s supports only 'l2' or 'none' penalties, \"\n","ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n","\n","  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"]},{"data":{"text/plain":["GridSearchCV(cv=5, estimator=LogisticRegression(),\n","             param_grid={'C': array([1.e-03, 1.e-02, 1.e-01, 1.e+00, 1.e+01, 1.e+02, 1.e+03]),\n","                         'penalty': ['l1', 'l2']})"]},"execution_count":33,"metadata":{},"output_type":"execute_result"}],"source":["param_grid = {\"C\":np.logspace(-3,3,7), \"penalty\":[\"l1\",\"l2\"]}\n","\n","grid = GridSearchCV(LogisticRegression(), param_grid, cv=5)\n","grid.fit(X_train, y_train)"]},{"cell_type":"code","execution_count":34,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["best score for Linear Regression 0.7644805194805195\n","best hypermarameters for Linear Regression {'C': 0.001, 'penalty': 'l2'}\n"]}],"source":["print(\"best score for Linear Regression\", grid.best_score_)\n","print(\"best hypermarameters for Linear Regression\", grid.best_params_)"]},{"cell_type":"code","execution_count":35,"metadata":{},"outputs":[{"data":{"text/plain":["0.7"]},"execution_count":35,"metadata":{},"output_type":"execute_result"}],"source":["LR_model = grid.best_estimator_\n","LR_model.score(X_test, y_test)"]},{"cell_type":"markdown","metadata":{},"source":["# Model accuracy"]},{"cell_type":"code","execution_count":36,"metadata":{},"outputs":[],"source":["from sklearn.metrics import jaccard_score\n","from sklearn.metrics import f1_score\n","from sklearn.metrics import log_loss"]},{"cell_type":"code","execution_count":37,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["KNN Jaccard : 0.7\n","KNN F1-score : 0.5764705882352941\n"]}],"source":["y_pred = kNN_model.predict(X_test)\n","print(\"KNN Jaccard :\", jaccard_score(y_test, y_pred, average=None)[1])\n","print(\"KNN F1-score :\", f1_score(y_test, y_pred, average='weighted'))"]},{"cell_type":"code","execution_count":38,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["Decision Tree Jaccard : 0.7\n","Decision Tree F1-score : 0.5764705882352941\n"]}],"source":["y_pred = DT_model.predict(X_test)\n","print(\"Decision Tree Jaccard :\", jaccard_score(y_test, y_pred, average=None)[1])\n","print(\"Decision Tree F1-score :\", f1_score(y_test, y_pred, average='weighted'))"]},{"cell_type":"code","execution_count":39,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["SVM Jaccard : 0.7\n","SVM F1-score : 0.5764705882352941\n"]}],"source":["y_pred = SVM_model.predict(X_test)\n","print(\"SVM Jaccard :\", jaccard_score(y_test, y_pred, average=None)[1])\n","print(\"SVM F1-score :\", f1_score(y_test, y_pred, average='weighted'))"]},{"cell_type":"code","execution_count":40,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["Linear regression Jaccard : 0.7\n","Linear regression F1-score : 0.5764705882352941\n","Linear regression LogLoss : 0.6209240963453982\n"]}],"source":["y_pred = LR_model.predict(X_test)\n","print(\"Linear regression Jaccard :\", jaccard_score(y_test, y_pred, average=None)[1])\n","print(\"Linear regression F1-score :\", f1_score(y_test, y_pred, average='weighted'))\n","\n","y_pred = LR_model.predict_proba(X_test)\n","print(\"Linear regression LogLoss :\", log_loss(y_test, y_pred))"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":[]}],"metadata":{"kernelspec":{"display_name":"Python 3.8.3 ('base')","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.3"},"vscode":{"interpreter":{"hash":"c871e2a932bd293b7fdb2227a0ba76194ca504637e3cb134bdc8ad50d4ee12ae"}}},"nbformat":4,"nbformat_minor":1}
